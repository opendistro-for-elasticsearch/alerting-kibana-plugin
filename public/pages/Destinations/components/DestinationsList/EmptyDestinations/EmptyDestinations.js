/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiButton, EuiEmptyPrompt, EuiText } from '@elastic/eui';

import { APP_PATH } from '../../../../../utils/constants';
import { PLUGIN_NAME } from '../../../../../../utils/constants';

const filterText =
  'There are no destinations matching your applied filters. Reset your filters to view all destinations.';
const emptyText = 'There are no existing destinations. Add a destination.';
const createDestinationButton = (
  <EuiButton fill href={`${PLUGIN_NAME}#${APP_PATH.CREATE_DESTINATION}`}>
    Add destination
  </EuiButton>
);

const resetFiltersButton = resetFilters => (
  <EuiButton fill onClick={resetFilters}>
    Reset filters
  </EuiButton>
);

const propTypes = {
  isFilterApplied: PropTypes.bool.isRequired,
  onResetFilters: PropTypes.func.isRequired,
};

const EmptyDestinations = ({ isFilterApplied, onResetFilters }) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={
      <EuiText>
        <p>{isFilterApplied ? filterText : emptyText}</p>
      </EuiText>
    }
    actions={isFilterApplied ? resetFiltersButton(onResetFilters) : createDestinationButton}
  />
);

EmptyDestinations.propTypes = propTypes;

export default EmptyDestinations;
