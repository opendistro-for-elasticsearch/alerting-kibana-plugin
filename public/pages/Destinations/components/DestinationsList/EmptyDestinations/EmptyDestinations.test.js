/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { mount, render } from 'enzyme';
import EmptyDestinations from './EmptyDestinations';

describe('<EmptyDestinations />', () => {
  test('should render empty destinations message', () => {
    const wrapper = render(
      <EmptyDestinations isFilterApplied={false} onResetFilters={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render no results for filter criteria', () => {
    const wrapper = render(<EmptyDestinations isFilterApplied onResetFilters={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should call reset Filters callback on click of Reset Filters Button', () => {
    const handleResetFilter = jest.fn();
    const wrapper = mount(<EmptyDestinations isFilterApplied onResetFilters={handleResetFilter} />);
    // Simulate Reset button click Click
    wrapper.find('button').simulate('click');
    expect(handleResetFilter).toHaveBeenCalledTimes(1);
  });
});
