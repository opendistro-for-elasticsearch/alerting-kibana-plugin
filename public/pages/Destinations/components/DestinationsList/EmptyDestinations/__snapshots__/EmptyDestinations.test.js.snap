// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyDestinations /> should render empty destinations message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no existing destinations. Add a destination.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <a
    class="euiButton euiButton--primary euiButton--fill"
    href="opendistro-alerting#/create-destination"
    rel="noreferrer"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Add destination
      </span>
    </span>
  </a>
</div>
`;

exports[`<EmptyDestinations /> should render no results for filter criteria 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width:45em"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no destinations matching your applied filters. Reset your filters to view all destinations.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <button
    class="euiButton euiButton--primary euiButton--fill"
    type="button"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Reset filters
      </span>
    </span>
  </button>
</div>
`;
